src/Ads/AdsRecommendationsService.php

```diff
@@ -0,0 +1,188 @@
+		$query = $this->container->get( AdsRecommendationsQuery::class );
+
+		if ( '' === $type ) {
+			$type = 'IMPROVE_PERFORMANCE_MAX_AD_STRENGTH';
+		}
+
+		// Filter by type if valid.
+		if ( $type === 'IMPROVE_PERFORMANCE_MAX_AD_STRENGTH' ) {
+			$query->where( 'recommendation_type', $type );
+		} else {
+			// If type is not valid, return an empty array.
+			return [];
+		}
+
+		if ( $id ) {
+			$query->where( 'recommendation_id', $id );
+		}
+
+		$result = $query->get_results();
+
+		$recommendations = [];
+
+		foreach ( $result as $item ) {
+			$recommendations[] = [
+				'id'              => (int) ( $item['recommendation_id'] ?? 0 ),
+				'type'            => $item['recommendation_type'] ?? '',
+				'resource_name'   => $item['recommendation_resource_name'] ?? '',
+				'campaign_id'     => (int) ( $item['recommendation_campaign_id'] ?? 0 ),
+				'campaign_name'   => $item['recommendation_campaign_name'] ?? '',
+				'campaign_status' => $item['recommendation_campaign_status'] ?? '',
+				'customer_id'     => (int) ( $item['recommendation_customer_id'] ?? 0 ),
+				'last_synced'     => isset( $item['recommendation_last_synced'] )
+					? gmdate( 'c', strtotime( $item['recommendation_last_synced'] ) )
+					: null,
+			];
+		}
+
+		// If no recommendations found, return an empty array.
+		if ( empty( $recommendations ) ) {
+			return [];
+		}
+		return $recommendations;
+	}
+
+	/**
+	 * Retrieves recommendations from the Google Ads API.
+	 *
+	 * @param array $args Query arguments.
+	 * @return array Array of recommendations.
+	 *
+	 * @throws Exception If the recommendations data can't be retrieved.
+	 */
+	public function get_google_recommendations( $args ): array {
+		try {
+			$response = ( new GoogleAdsRecommendationsQuery( $args ) )
+			->set_client( $this->client, $this->options->get_ads_id() )
+			->get_results();
+
+			$result = [];
+			foreach ( $response->iterateAllElements() as $row ) {
+				if ( ! $row->hasRecommendation() ) {
+					continue;
+				}
+
+				$recommendation = $row->getRecommendation();
+
+				// Skip if recommendation is not valid.
+				if ( ! $recommendation instanceof Recommendation ) {
+					continue;
+				}
+
+				$campaign = $row->getCampaign();
+				$customer = $row->getCustomer();
+
+				$result[] = [
+					/**
+					 * Note: The 'id' field below refers to the Recommendation resource's ID property, not the field name itself.
+					 * Reference: https://github.com/googleads/google-ads-php/blob/main/src/Google/Ads/GoogleAds/V18/Resources/Recommendation.php#L25-L30
+					 *
+					 * We use the static name for the recommendation type instead of `$recommendation->getType()`
+					 * to ensure consistency and avoid potential issues with dynamic values or API changes.
+					 */
+					'recommendation_type'            => 'IMPROVE_PERFORMANCE_MAX_AD_STRENGTH',
+					'recommendation_resource_name'   => method_exists( $recommendation, 'getResourceName' ) ? $recommendation->getResourceName() : '',
```

Comment:
@mukeshpanchal27 we also need to include the `recommendation_id` which we create from the resource name.